LNP-BP/rgb-node
# [WIP] Sqlite Database engine for asset cache

This closes [Issue#58](https://github.com/LNP-BP/rgb-node/issues/58). This is WIP and intended for initial review and suggestions for possible modifications.

**Note:** 
Two test cases are added to test the working of an asset db. These tests are ignored by default as they will break travis build.
To test the DB do the followings: 

- Install diesel-cli by running `cargo install diesel-cli`
- Setup an environment variable pointing at the rgb datadir `export DATABASE_URL=~/.rgb`
- Manually remove `#[ignore]` flag from tests in `src/contract/fungible/cache/sql.rs`.
- run the testing script in `test/test_db.sh`.

This should run the tests, and if all goes well they should pass.

**Rough Edges**

- `Chain` variants are not fully covered. The database only supports `MainNet` and `Testnet3` for now as they can be encoded via text. The rest of the variants need more expressive table types or probably new tables to handle them. Suggestions are welcome. At least if we can do `Regtest` it will be usable for experimenting with real rgb-node.

**Possible Improvements**  
Once this PR gets finalized it would better to integrate Travis to run these tests. They have to be handled separately from regular tests as it involves running `diesel-cli` and setting up env variable.
